### DISCUSSION TITLE
styled and cs in typescript

### DISCUSSION THREAD

**Discussion by smalia2001** [ben-rogerson/twin.macro#311] [Q&A] [answered]
Hi, I saw that you changed the `css` dom attribute to `cs`, and `styled` is not declared in `index.d.ts`, is that on purpose? I can submit a pr for it, but I don't really know what are you planning 😄 , Amazing job btw!!!

**Comment by ben-rogerson** [comment, marked as answer]
Hey there!

I've added the cs prop (and it's types) but it's different to the css prop and allows the short css syntax to be written in there. It also works with the css prop, so you can use both props at the same time if you need to.

eg:

<div cs="md:maxWidth[calc(100vw - 2em)]" css={`color: red;`} />
The types for styled come from your chosen css-in-js library, if you want to set that up check out one of the example readmes for the setup.

**Reply by smalia2001** [reply]
Ok,I Installed 
```json
"@emotion/core": "^11.0.0",
"@emotion/react": "^11.1.4",
"@emotion/styled": "^11.0.0"
```
And I still get:
```
import css
Module '"../node_modules/twin.macro/types"' has no exported member 'css'. Did you mean to use 'import css from "../node_modules/twin.macro/types"' instead?
```
when I try to 
```tsx
import tw, { css } from 'twin.macro'
```
same for `styled`, what should I do differently?

**Reply by ben-rogerson** [reply]
Did you check the typescript docs on those example readmes?

![typescript-docs](https://user-images.githubusercontent.com/21288568/106430451-b7c90a80-64bb-11eb-82f5-bf800046fcff.gif)

**Reply by smalia2001** [reply]
My bad, didn't see it. 
thanks!!

**Reply by srkrish2** [reply]
Where can I find the readme shown above @ben-rogerson ? I'm running into the same issue but can't find the resource.

**Reply by ben-rogerson** [reply]
@srkrish2 Only some of the examples have the typescript section (yep I'll be improving this :) )

Here's some links:
[emotion](https://github.com/ben-rogerson/twin.examples/tree/master/vite-emotion-typescript#add-typescript-types) / [styled-components](https://github.com/ben-rogerson/twin.examples/tree/master/vite-styled-components-typescript#add-typescript-types)

**Comment by therealgilles** [comment]
Is there a reason why the cs prop is nowhere to be found in the documentation?

**Reply by ben-rogerson** [reply]
Never quite gained enough traction as a feature, but it's being deprecated soon so not worth adding it now
